### PR TITLE
[feature](serde) support presto compatible output format

### PR DESCRIPTION
**Pull request by morningman**
the output format of some data types are different between Presto/Trino and Doris,
especially for complex type such as array, map and struct.
When user migrate from Presto to Doris, they expect the same format so that they
don't need to modify their business code.

This PR mainly changes:

1. Add a new session variable `serde_dialect`
    Default is `doris`, options are `presto` or `trino`. If set to presto or trino,
    the output format returned to MySQL client of some datatypes will be changed:

    - Array
        Doris: `["abc", "def", "", null]`
        Presto: `[abc, def, , NULL]`

    - Map
        Doris: `{"k1":null, "k2":"v3"}`
        Presto: `{k1=NULL, k2=v3}`

    - Struct
        Doris: `{"s_id":100, "s_name":"abc , "", "s_address":null}`
        Presto: `{s_id=100, s_name=abc , ", s_address=NULL}`
 
2. Change the output format of struct type

    Remove the space after `:`

    - Before: `{"s_id": 100, "s_name": "abc , "", "s_address": null}`
    - After: ``{"s_id":100, "s_name":"abc , "", "s_address":null}``

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

File: be/src/common/serde_info.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+namespace doris {
+
+// The description of serde format.
+// Used for the string format of result return to MySQL client.
+struct SerdeInfo {
```
Comment (amorynan) [inline be/src/common/serde_info.h:24]: maybe this struct can put into DataTypeSerDe.FormatOptions.mysql.serdeInfo ?

Comment (amorynan) [reply be/src/common/serde_info.h:24]: and we have some option definition can be reuse ?

Comment (morningman) [reply be/src/common/serde_info.h:24]: You are right..

**Comment by morningman** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Review by BePPPower — APPROVED** [review]

**Review by amorynan — APPROVED** [review]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**